src/odemis/driver/piezomotor.py

```diff
@@ -0,0 +1,733 @@
+        """
+        resp = self._sendCommand(b'X%dJ' % axis)
+        if int(resp) == 0:  # 1 if finished, 0 while moving
+            return True
+        else:
+            return False
+
+    def startIndexMode(self, axis):
+        """
+        Enters index mode.
+        """
+        self._sendCommand(b'X%dN4' % axis)
+
+    def moveToIndex(self, axis):
+        """
+        Move towards the index until it's found.
+        """
+        self._sendCommand(b'X%dI-16' % axis)
+
+    def getIndexStatus(self, axis):
+        """
+        Return a short description of the index status, e.g. "1,132.,indexed". The dot after the logged
+        index position indicates that the position was logged since the last report. 'indexed'
+        means the position has been reset at index.
+        """
+        # Check if referencing was successful
+        return self._sendCommand(b'X%dN?' % axis)
+
+    def _sendCommand(self, cmd):
+        """
+        :arg cmd (bytes): command to be sent to the hardware
+        :returns (bytes): response
+        """
+        cmd += EOL
+        with self._ser_access:
+            logging.debug("Sending command %s", to_str_escape(cmd))
+            self._serial.write(cmd)
+
+            resp = b""
+            while resp[-len(EOL):] != EOL:
+                try:
+                    char = self._serial.read()
+                except IOError:
+                    logging.warn("Failed to read from PMT Control firmware, "
+                                 "trying to reconnect.")
+                    if self._recovering:
+                        raise
+                    else:
+                        self._tryRecover()
+                        # don't send command again
+                        raise IOError("Failed to read from PMT Control firmware, "
+                                      "restarted serial connection.")
+                if not char:
+                    raise IOError("Timeout after receiving %s" % to_str_escape(resp))
+                else:
+                    resp += char
+            logging.debug("Received response %s", to_str_escape(resp))
+
+            # Check response (command should be echoed back)
+            if not resp.startswith(cmd[:-len(EOL)]):
+                raise IOError("Response starts with %s != %s", resp[:len(cmd)], cmd)
+            if b"_??_" in resp:
+                raise ValueError("Received response %s, command %s not understood." % (resp, cmd))
+            if b"!" in resp:
+                raise PMDError(0, resp)
+            # Format:
+            #    * for query with response: <cmd>:<ret><EOL> (will return <ret>)
+            #    * for set command without response: <cmd><EOL> (will return b"")
+            return resp[len(cmd) + 1 - len(EOL):-len(EOL)].decode("latin1")
+
+    def _tryRecover(self):
+        self._recovering = True
+        self.state._set_value(HwError("USB connection lost"), force_write=True)
```

Comment:
As this message is shown to the user (nowadays), it's good to let the user she/he shouldn't worry yet. So I'd recommend something like "Connection lost, reconnecting...".